### ISSUE TITLE
Cloning git repo causes visual studio 2019 to indefinitely hang

### ISSUE THREAD

**Issue by vsfeedback** [microsoft/PTVS#5953]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/846215/cloning-git-repo-causes-visual-studio-2019-to-inde.html)._

---
[regression] [worked-in:The version before 16.13.10]
Every time I clone from dev.azure.com/office, visual studio just hangs indefinitely and I have to kill it to get it back into a working state. I can't reproduce steps for this session and it will just hang and I can't file feedback.

---
#### Visual Studio Feedback System on 12/5/2019, 07:15 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by huguesv** [comment]
We have a dump file in the internal bug 1032969, see that for additional details. Call stack for the hang is:
```
Top of callstack:

> System.dll!System.ComponentModel.Component.Dispose(bool disposing) Line 183 C#_LSP
  System.dll!System.IO.FileSystemWatcher.Dispose(bool disposing) Line 649 C#_LSP​
  System.dll!System.ComponentModel.Component.Dispose() Line 120 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.WorkspaceInterpreterFactoryProvider.InitializeWorkspace(Microsoft.PythonTools.Interpreter.IPythonWorkspaceContext workspace) Line 121 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.WorkspaceInterpreterFactoryProvider.EnsureInitialized() Line 91 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.WorkspaceInterpreterFactoryProvider.InterpreterFactoriesChanged.add(System.EventHandler value) Line 391 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.InterpreterRegistryService.EnsureFactoryChangesWatched() Line 81 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.InterpreterRegistryService.InterpretersChanged.add(System.EventHandler value) Line 70 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.InterpreterOptionsService.InitializeDefaultInterpreterWatcher() Line 75 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.InterpreterOptionsService.DefaultInterpreterChanged.add(System.EventHandler value) Line 266 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.PythonWorkspaceContext.Initialize() Line 112 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.PythonWorkspaceContextProvider.InitializeCurrentContext() Line 116 C#_LSP​
  Microsoft.PythonTools.VSInterpreters.dll!Microsoft.PythonTools.Interpreter.PythonWorkspaceContextProvider.OnActiveWorkspaceChanged(object sender, System.EventArgs e) Line 88 C#_LSP​
```

**Comment by r3m0t** [comment]
This happened for me when I don't have any Python environments installed

**Comment by huguesv** [comment]
Looks like this is the same deadlock as https://github.com/microsoft/PTVS/issues/6203, which is now fixed (for 16.8 Preview 3)
